domains/authorization/Authorization.API/API/Authorization/[Features]/Internal/GetServiceProviderTermsForOrganizationQuery.cs

```diff
@@ -0,0 +1,49 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Authorization.Exceptions;
+using API.Repository;
+using EnergyOrigin.Domain.ValueObjects;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Authorization._Features_.Internal;
+
+public record GetServiceProviderTermsForOrganizationQuery(string OrgCvr) : IRequest<bool>;
+
+
+public class GetServiceProviderTermsQueryHandler(IOrganizationRepository organizationRepository, IServiceProviderTermsRepository serviceProviderTermsRepository)
+    : IRequestHandler<GetServiceProviderTermsForOrganizationQuery, bool>
+{
+    public async Task<bool> Handle(GetServiceProviderTermsForOrganizationQuery request, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrWhiteSpace(request.OrgCvr))
+        {
+            throw new InvalidConfigurationException("Organization CVR is required.");
+        }
+
+        var organizationCvr = Tin.Create(request.OrgCvr);
+
+        var organization = await organizationRepository.Query()
+            .FirstOrDefaultAsync(o => o.Tin == organizationCvr, cancellationToken);
+
+        if (organization == null)
+        {
+            throw new InvalidConfigurationException("Organization not found.");
```

Comment:
NotFoundException, or EntityNotFound. Automatic returns 404 in apis and used other places.

Reply:
Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1605/commits/4950dc0e89bc222374cf16496ee20b9da9dfd2af